packages/node/src/plugins/segmentio/publisher.ts

```diff
@@ -183,44 +186,38 @@ export class Publisher {
       this._closeAndFlushPendingItemsCount -= batch.length
     }
     const events = batch.getEvents()
-    const payload = JSON.stringify({ batch: events })
     const maxAttempts = this._maxRetries + 1
 
     let currentAttempt = 0
     while (currentAttempt < maxAttempts) {
       currentAttempt++
 
       let failureReason: unknown
-      const [signal, timeoutId] = abortSignalAfterTimeout(
-        this._httpRequestTimeout
-      )
       try {
-        const requestInit = {
-          signal: signal,
```

Comment:
Did you have a specific fetch replacement library in mind that doesn't support 'signal' here?  I have questions about simultaneously allowing access to tinker with our internals and passing around non-standard objects.

Reply:
All fetch libraries support AbortSignal, but not all HTTP Libraries do (e.g Request).
 
That's why signal  is only present in the HTTPFetch interface.

Reply:
I just wanted to know what the impetus was.  Figured you had a reason for moving some of the fiddly bits into the user replaceable thing that they then have to repeat.